lib/crypto_env_var/cipher.rb

```diff
@@ -2,16 +2,146 @@
 
 module CryptoEnvVar
   class Cipher
+    SEPARATOR = "--".freeze
+    DIGEST_SEPARATOR = "----".freeze
```

Comment:
Instead of dealing with structure encoding ourselves - how about keeping them as Ruby data structures most of the way, and encoding with Messagepack+Base64 at the last minute?
That'd avoid some error-prone dancing around with join/split?

Reply:
I'm doing a similar PR atm, and I used null characters as the separators. The msgpack solution looks better actually.